### PR TITLE
Fix: ensure name normalization for schema methods

### PR DESCRIPTION
**Pull request by GeorgeSittas**
This PR aims to make the normalization of names in `MappingSchema` more robust. We're already normalizing names, as introduced in https://github.com/tobymao/sqlglot/pull/891, but that's only limited to the schema constructor. All of the exposed methods (`add_table`, `column_names`, `get_column_type`) don't normalize their input arguments.

cc: @tobymao @barakalon

### CODE REVIEW

File: sqlglot/schema.py [hunk 1]
```diff
@@ -206,84 +262,40 @@ def _normalize(self, schema: t.Dict) -> t.Dict:
...
 
     def _normalize_name(self, name: str) -> str:
         try:
-            identifier: t.Optional[exp.Expression] = sqlglot.parse_one(
-                name, read=self.dialect, into=exp.Identifier
+            identifier = t.cast(
+                exp.Identifier, sqlglot.parse_one(name, read=self.dialect, into=exp.Identifier)
             )
         except:
-            identifier = exp.to_identifier(name)
```
Comment (tobymao) [inline sqlglot/schema.py:239]: why was this changed? what case triggered the need to change this

Comment (GeorgeSittas) [reply sqlglot/schema.py:239]: There wasn't a case that triggered this change, but it felt reasonable to return `name` here if we are unable to parse it into an identifier. I guess there are two cases to consider here. If `name` can't be parsed and it is converted `to_identifier`, then the resulting `exp.Identifier` is:

- quoted, in which case we'd return `name` (this behaviour is preserved in this PR), or
- not quoted, so we'd return `name.lower()`.

How we deal with the latter case is the only difference here. The question is: why should we convert to an identifier and call `.lower` to begin with, if the name couldn't be parsed as one? Seemed more logical to just return `name`, but I might be wrong. Happy to revert this change if it looks risky.

FWIW, in my original PR Barak had suggested using `parse_one` here too (https://github.com/tobymao/sqlglot/pull/891#discussion_r1062872629), but I can see how `to_identifier` might make this more conservative.
